Lib/extractor/__init__.py

```diff
@@ -75,3 +75,65 @@ def extractUFO(
         raise ExtractorError(
             "There was an error reading the %s file." % format
         )
+
+
+def cmdline():
+    """
+    Extract one ore more fonts to UFO. Installed as command line script
+    `extractufo`.
+
+    Usage: extractufo font [font ...]
+    """
+    from sys import exit
+    from argparse import ArgumentParser
+
+    parser = ArgumentParser(
+        description="Extract data from font binaries and build UFO objects from them.",
+        epilog="Each resulting UFO will be saved as FONT_FILE.ufo(z) in the same directory as the original FONT_FILE.",
+    )
+    parser.add_argument('FONT_FILE', help='Input font path', nargs="+")
+    parser.add_argument('-m', '--ufo-module', choices=['ufoLib2', 'defcon'], help='Select the default library for writing UFOs (default: autodetect, prefer ufoLib2)')
+    parser.add_argument('-z', '--zip', action="store_true", help="Output UFO ZIP")
+    parser.add_argument('-o', '--overwrite', action="store_true", help="Overwrite output location if exists")
+
+    args = parser.parse_args()
+    if args.ufo_module is None:
+        try:
+            from ufoLib2 import Font
+            print("Will use ufoLib2 for UFO output.")
+        except ImportError:
+            try:
+                from defcon import Font
+                print("Will use defcon for UFO output.")
+            except ImportError:
+                print("Either ufoLib2 or, alternatively, defcon library is required to run this command.\nPlease install one of them.")
+                exit(1)
+    elif args.ufo_module == 'ufoLib2':
+        try:
+            from ufoLib2 import Font
+        except ImportError:
+            print("Can't find ufoLib2 installed. Please install it or specify a different UFO library.")
+            exit(1)
+    else:
+        try:
+            from defcon import Font
+        except ImportError:
+            print("Can't find defcon installed. Please install it or specify a different UFO library.")
+            exit(1)
+
+    structure="zip" if args.zip else "package"
+    had_write_errors = False
+    for font_path in args.FONT_FILE:
+        ufo_path = f"{font_path}.ufo" if not args.zip else f"{font_path}.ufoz"
+        print(f"Extracting {ufo_path}... ", end="")
+        ufo = Font()
+        extractUFO(font_path, ufo)
+        try:
+            ufo.save(ufo_path, overwrite=args.overwrite, structure=structure)
```

Comment:
could just drop overwrite here for simplicity

Reply:
Okay, I've removed the option, and added an explicit check for file/dir existence. If it exists, that input file will not be converted.
I reckon that's better than having two different scenarios, right? 
Hm, maybe I should make a PR for defcon adding this option though... :laughing: